File: app/src/androidTest/java/com/games/memorygame/injection/ScoreDataSourceMock.java
package com.games.memorygame.injection;

import com.games.memorygame.model.Score;
import com.games.memorygame.persistence.ScoreDataSource;

import java.util.ArrayList;
import java.util.Date;
import java.util.GregorianCalendar;
import java.util.List;

import io.reactivex.Flowable;

public class ScoreDataSourceMock implements ScoreDataSource {

    private List<Score> scores = new ArrayList<>();

    @Override
    public Flowable<List<Score>> loadScores() {
        Date date = new GregorianCalendar(2017, 7, 27).getTime();
        for (int i = 0; i < 11; i++) {
            scores.add(new Score(String.valueOf(i), "User " + i, date, i));
        }
        return Flowable.just(scores);
    }

    @Override
    public void addScore(Score score) {
        scores.add(score);
    }

    @Override
    public void clearData() {
        scores.clear();
    }

}

File: app/src/androidTest/java/com/games/memorygame/injection/ScoreDataSourceMock.kt
package com.games.memorygame.injection

import com.games.memorygame.model.Score
import com.games.memorygame.persistence.ScoreDataSource

import java.util.ArrayList
import java.util.GregorianCalendar

import io.reactivex.Flowable

class ScoreDataSourceMock : ScoreDataSource {

    private val scores = ArrayList<Score>()

    override fun loadScores(): Flowable<List<Score>> {
        val date = GregorianCalendar(2017, 7, 27).time
        (0 until 10).forEach { scores.add(Score("User " + it, it, it.toString(), date)) }
        return Flowable.just(scores)
    }

    override fun addScore(score: Score) {
        scores.add(score)
    }

    override fun clearData() {
        scores.clear()
    }

}
